qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-renarde/.classpath

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" output="target/classes" path="src/main/java">
+		<attributes>
+			<attribute name="optional" value="true"/>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry excluding="**" kind="src" output="target/classes" path="src/main/resources">
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="src" output="target/test-classes" path="src/test/java">
+		<attributes>
+			<attribute name="test" value="true"/>
+			<attribute name="optional" value="true"/>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry excluding="**" kind="src" output="target/test-classes" path="src/test/resources">
+		<attributes>
+			<attribute name="test" value="true"/>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11">

```

Comment:
@angelozerr , I made the following change, and type hierarchy worked for me.

```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">

```

```
typeHierarchy.getAllSubtypes(type)

Application (not open) {key=Lrest/Application;} [in Application.java [in rest [in src/main/java [in qute-renarde]]]]
ControllerWithUser (not open) {key=Lio/quarkiverse/renarde/oidc/ControllerWithUser<TU;>;} [in ControllerWithUser.class [in io.quarkiverse.renarde.oidc [in lib/quarkus-renarde-oidc-1.0.0-SNAPSHOT.jar [in qute-renarde]]]]
Login (not open) {key=Lrest/Login;} [in Login.java [in rest [in src/main/java [in qute-renarde]]]]
Todos (not open) {key=Lrest/Todos;} [in Todos.java [in rest [in src/main/java [in qute-renarde]]]]
RenardeSecurityController (not open) {key=Lio/quarkiverse/renarde/oidc/impl/RenardeSecurityController;} [in RenardeSecurityController.class [in io.quarkiverse.renarde.oidc.impl [in lib/quarkus-renarde-oidc-1.0.0-SNAPSHOT.jar [in qute-renarde]]]]
```

Reply:
@fbricon , Occasionally, I see the following errors when I import qute-renarde into an Eclipse workspace :

```
Description	Resource	Path	Location	Type
The project was not built since its build path is incomplete. Cannot find the class file for javax.ws.rs.core.NewCookie. Fix the build path then try building this project	qute-renarde		Unknown	Java Problem
The type javax.ws.rs.core.NewCookie cannot be resolved. It is indirectly referenced from required .class files	Login.java	/qute-renarde/src/main/java/rest	line 1	Java Problem
```

The  above change resolves it. Could this be an Eclipse-specific (maybe m2e-specific) issue that's causing the failure to resolve ?